### PR TITLE
feature: Call Title and Subtitle

### PR DESCRIPTION
**Pull request by dhiraj**
## Purpose

* This PR adds support for Contoso to set a title and subtitle for a group call
* Settings activity has been modified to have two new text boxes to setup a call title and subtitle

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [x] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [x] UI Changes
  - [x] Screen captures included
  - [x] Tested for Light/Dark mode
  - [x] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Open the app
* Open settings
* Set up a title and subtitle in the new fields

## What to Check
Verify that the following are valid
* When a title is set, the call setup screen shows it, instead of "Setup"
* When a subtitle is set, the call setup screen shows it as expected

## Other Information
* This is being posted a public API change mainly because `CallCompositeLocalOptions` was modified to include two new `String` fields for the call title and subtitle
* Both Kotlin and Java launchers now _always_ send Local Options

## Screenshots
Screenshot_1659647191

Screenshot_1659647182

Screenshot_1659647113

Screenshot_1659647147

### CODE REVIEW

File: azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/models/CallCompositeLocalOptions.java [hunk 1]
```diff
@@ -33,14 +35,21 @@
...
+    private final String callSubTitle;
+
     /**
      * Create LocalSettings.
      *
      * @param participantViewData The {@link CallCompositeParticipantViewData};
      * @see CallCompositeParticipantViewData
      */
-    public CallCompositeLocalOptions(final CallCompositeParticipantViewData participantViewData) {
+    public CallCompositeLocalOptions(final CallCompositeParticipantViewData participantViewData,
```
Comment (pavelprystinka) [inline azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/models/CallCompositeLocalOptions.java:48]: How about we will add a default constructor that does not take any parameters and add settert to all fields?

Comment (dhiraj) [reply azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/models/CallCompositeLocalOptions.java]: @pavelprystinka  Fixed in d1ea113

**Review by Mrayyan — CHANGES_REQUESTED** [review]
Tested looks good. 👍 

Please review with Alex P on what to do with cases like empty/blank title and subtitle. Min or max length for title and subtitle as well as what happens when an emoji is passed in.

Also, please update the changelog.

@pavelprystinka Do we first get this signed off by ARB and then merge it or is ARB after merge?

**Comment by dhiraj** [comment]
> Please review with Alex P on what to do with cases like empty/blank title and subtitle. Min or max length for title and subtitle as well as what happens when an emoji is passed in.

When title is not specified we revert to current behavior and show "Setup" as the title. If a subtitle is specified, _without_ a title it is shown, under the current default title, Setup. I'll check with Alex P on min/max lengths for both the new fields. Thanks @Mrayyan 

(Screenshot with empty title and subtitle looks the same):
Screenshot_1659732185

**Comment by dhiraj** [comment]
@Mrayyan I've updated the CHANGELOG with a new Features section, please re-review, thanks!

**Review by Mrayyan — DISMISSED** [review]
Looks good 👍

File: azure-communication-ui/azure-communication-ui-demo-app/src/calling/java/com/azure/android/communication/ui/callingcompositedemoapp/launcher/CallingCompositeKotlinLauncher.kt [hunk 2]
```diff
@@ -38,7 +40,8 @@ class CallingCompositeKotlinLauncher(private val tokenRefresher: Callable<String
         showAlert: ((String) -> Unit)?,
     ) {
         initialize(callLauncherActivity.applicationContext)
-        val participantViewData = getParticipantViewData(callLauncherActivity.applicationContext)
+        val
```
Comment (Mrayyan) [inline azure-communication-ui/azure-communication-ui-demo-app/src/calling/java/com/azure/android/communication/ui/callingcompositedemoapp/launcher/CallingCompositeKotlinLauncher.kt:43]: nitpicking here, you can bring this back to a single line.

**Comment by dhiraj** [comment]
After syncing up with iOS and UI design, I've updated the color of the main titlebar from full black (#000000) to almost black (#212121) and added a new color for subtitle as well. Updated screenshots:
Screenshot_1660597375
Screenshot_1660597388

**Review by Mohtasim — DISMISSED** [review]

**Comment by dhiraj** [comment]
Customized textSize styling on actionbar to reduce title and subtitle to match Figma mockups.

Updated Screenshots (note how subtitle accommodates more words now):
Screenshot_1660685206
Screenshot_1660685212

File: azure-communication-ui/azure-communication-ui-demo-app/src/calling/java/com/azure/android/communication/ui/callingcompositedemoapp/launcher/CallingCompositeKotlinLauncher.kt (hunk 2 above)
Comment (dhiraj) [reply azure-communication-ui/azure-communication-ui-demo-app/src/calling/java/com/azure/android/communication/ui/callingcompositedemoapp/launcher/CallingCompositeKotlinLauncher.kt]: You're right, fixed in 8853acd

**Review by Mrayyan — APPROVED** [review]

**Review by phileo99 — APPROVED** [review]
👍

**Review by ShaunaSong — APPROVED** [review]

**Pull request merged**